assets/src/components/editing/elements/webpage/webpageActions.tsx

```diff
@@ -14,75 +13,50 @@ export type WebpageCreationProps = {
   onChange: (src: string) => void;
   onEdit: (src: string) => void;
 };
-const WebpageCreation = (props: WebpageCreationProps) => {
-  const [src, setSrc] = useState('');
 
-  return (
-    <div>
-      <div className="form">
-        <label>Enter the webpage URL:</label>
-        <input
-          type="text"
-          value={src}
-          onChange={(e) => {
-            props.onChange(e.target.value);
-            setSrc(e.target.value);
-          }}
-          onKeyPress={(e: any) => Settings.onEnterApply(e, () => props.onEdit(src))}
-          className="form-control mr-sm-2"
-        />
-        <div className="mb-2">
-          <small>e.g. https://www.wikipedia.org</small>
-        </div>
-      </div>
-    </div>
-  );
-};
-
-export function selectWebpage(): Promise<string | null> {
+export function selectWebpage(projectSlug: string): Promise<Webpage | null> {
   return new Promise((resolve, _reject) => {
-    const selected: { src: null | string } = { src: null };
-
-    const mediaLibrary = (
-      <Modal
-        title="Insert Webpage"
-        onOk={() => {
-          dismiss();
-          resolve(selected.src ? selected.src : '');
-        }}
-        onCancel={() => dismiss()}
-      >
-        <WebpageCreation
-          onEdit={(src: string) => {
-            dismiss();
-            resolve(src);
-          }}
-          onChange={(src: string) => {
-            selected.src = src;
-          }}
-        />
-      </Modal>
+    const initial = Model.webpage('');
+
+    const onDone = (webpage: Partial<Webpage>) => {
+      resolve({
+        ...initial,
+        ...webpage,
+      });
+      dismiss();
+    };
+
+    const onCancel = () => {
+      resolve(null);
+      dismiss();
+    };
+
+    display(
```

Comment:
inserting an iframe behaves a bit different now.

Before it would ask for a url, now it uses the entire iframe settings modal so you can choose from media library and also set the alt text on creation.